push-build.sh

```diff
@@ -185,8 +186,19 @@ if ((FLAGS_federation)); then
   common::stepheader PUSH FEDERATION
   ############################################################################
   logecho -n "Push federation images: "
-  # FEDERATION_PUSH_REPO_BASE should be set by the calling job (yaml)
-  logrun -s ${KUBE_ROOT}/federation/develop/push-federation-images.sh
+  GCLOUD_PROJECT=$($GCLOUD config list project --format 'value(core.project)')
+  if [[ -z "${GCLOUD_PROJECT}" ]]; then
+    logecho "No account authorized through gcloud. So using FEDERATION_PUSH_REPO_BASE to push images"
+    KUBE_DOCKER_REGISTRY=${FEDERATION_PUSH_REPO_BASE}
```

Comment:
Agree. but the `make release` is called from a [script](https://github.com/kubernetes/kubernetes/blob/master/hack/jenkins/build.sh#L56) from `github.com/kubernetes/kubernetes`v repo, making all this synchronization difficult.